### PR TITLE
enhance the code block

### PR DESCRIPTION
**Pull request by starsareintherose**
try to fix it, I am not so familiar with it, so if any problems, better to point them out.

### CODE REVIEW

File: sourceview/markdown.lang [hunk 1]
```diff
@@ -319,7 +319,21 @@
...
+    <!-- Examples:
+         Here's a literal HTML tag: ``<p>``.
+         ``The grave accent (`) is used in Markdown to indicate a code span.``
+         ``Here's another code span containing enter and four space before every lines of coded content
+    -->
+    <context id="2-backticks-code-span" class="no-spell-check mono" style-ref="code">
+      <match>(?&lt;!`)``[^`]+((`|`{3,})[^`]+)*``(?!`)</match>
+    </context>
 
+    <context id="3-backticks-code-span" class="no-spell-check mono" style-ref="code">
```
Comment (sp1ritCS) [inline sourceview/markdown.lang:332]: four-space codeblocks have been removed, due to them interfering with enumerate/itemize lists.

See https://github.com/blackhole89/notekit/issues/32 & https://github.com/blackhole89/notekit/pull/77 for more info.

Comment (starsareintherose) [reply sourceview/markdown.lang:332]: it's not wise to rm it, it contains two enter and four space, I can't say it's conflict

**Comment by blackhole89** [comment]
There are a number of issues here. 

* the two rules you introduced are named the same as already existing rules (which overwrites those rules)
*  the names are therefore also no longer descriptive (the four-space indentation is not a "3-backtick-code-span"). 
*  this does not appear to solve the problem that we can't parse four-space-indented blocks as code unless we are sure that they are not part of any other block-level indented construct, such as bulleted lists.

**Comment by starsareintherose** [comment]
> There are a number of issues here.
> 
> * the two rules you introduced are named the same as already existing rules (which overwrites those rules)
> * the names are therefore also no longer descriptive (the four-space indentation is not a "3-backtick-code-span").
> * this does not appear to solve the problem that we can't parse four-space-indented blocks as code unless we are sure that they are not part of any other block-level indented construct, such as bulleted lists.

okay, and could you show me some documents, I write these codes by guessing, I don't know about its details.

**Comment by blackhole89** [comment]
Documents for the language spec: [here](http://getfr.org/pub/dragonfly-release/usr-local-share/doc/gtksourceview-3.0/lang-reference.html)

Markdown spec: [Github's version is probably the most legible](https://github.github.com/gfm/)

**Pull request closed without merging**